enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+
+The following items must be satisfied in this phase:
+
+- the installer must allow one item of machine pool type `edge` (`compute[?name=="edge"]`)
+- the installer must extract Local Zone subnets from the subnet IDs
+  on `platform.aws.subnets`, creating one MachineSet manifest for each zone.
+- the `compute[?name=="edge"]` must create MachineSets manifests on Local Zone
+  subnets with:
+  - labels: `node-role.kubernetes.io/edge=''`, `location=local-zone`, and
+    `zone_group=zone_group_name`
+  - and taint `NoSchedule` matching the label `node-role.kubernetes.io/edge`
+- the MachineSet should be created when any Local Zone subnet is added on
+  `platform.aws.subnets`, even if compute pool was not defined on `compute[?name=="edge"]`
+
+> Note: Local Zones subnet scope. See section "Public subnet and outbound traffic"
+- only `public` subnets are allowed for Local Zone subnets ID added on `platform.aws.subnets`
+- the Local Zone type should not be allowed on the `worker` compute pool
+  `compute[?name=="worker"].platform.aws.zones`.
+
+- the default or user-provided (`compute[?name=="edge"].platform.aws.type` or
+  `platform.aws.defaultMachinePlatform.type`) instance type for the machine
+  pool `compute[?name=="edge"]` should be validated on the EC2 offering for the
+  target availability zone
+
+> Note: Currently the resource types are limited. The intersection between zones is c5d.2xlarge and gp2
+- the default `edge` machine pool instance type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `c5d.2xlarge`.
+- the default `edge` machine pool EBS type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `gp2`.
+
+#### Examples configuration for user-provided network
+
+Those are two examples of a user-provided network when installing a cluster
+in existing VPC:
+
+1) `install-config.yaml` with Local Zone subnets, and empty `edge` machine pool
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+2) `install-config.yaml` with Local Zone subnets, and custom instance type
+
+> Note: the instance type `c5d.2xlarge` common between locations
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+compute:
+- name: edge
+  platform:
+    aws:
+      type: c5d.2xlarge
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+    - subnetID-us-east-1-mia-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+3) `install-config.yaml` with Local Zone subnets, and custom volume type
+
+> Note: currently only `us-west-2` supports `gp3`
```

Comment:
Is this exposed through an API so we can use it to validate this kind of restriction beforehand?

Reply:
This topic was moved to a section ["Volume type"](https://github.com/openshift/enhancements/pull/1232/commits/fdbdba4c38f3b17c5159b4f5206a6a3671bc74ce#diff-f2590494c9138551ab5b3d5632d11170fe8742a128256f9e1d8ebc63baa19f41R216). Thanks!